modules/core/shared/src/main/scala/org/legogroup/woof/EnclosingClass.scala

```diff
@@ -0,0 +1,20 @@
+package org.legogroup.woof
+
+import scala.annotation.tailrec
+
+case class EnclosingClass(fullName: String, lineLength: Int = 80):
+  def printableName: String = reduceNameLength(fullName)
+
+  @tailrec
+  private def reduceNameLength(name: String, chomps: Int = 0): String = {
```

Comment:
I'm wondering if there's a slightly more functional way to write this one -- it's a mouthful to read like this

Reply:
```scala
  private def reduceNameLength(name: String): String =
    List
      .unfold(name.split('.').toList) {
        case Nil                                                   => none
        case last :: Nil                                           => (last, Nil).some
        case parts @ x :: xs if parts.mkString.length > lineLength => (x.take(1), xs).some
        case x :: xs                                               => (x, xs).some
      }
      .mkString(".")
```